### ISSUE TITLE
[codegen] Make sure overlay types are updated to match recent codegen

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-kubernetes#1764]
It's been awhile since the overlay templates were updated, so make sure that they match any recent codegen changes.

**Comment by EronWright** [comment]
@lblackstone do I understand correctly that this issue is about updating the "well-known" kinds, deprecations, replacement hints, etc? And not about overlay components.

**Comment by lblackstone** [comment]
No, this was specific to the overlay components. For example, the Python overlays for `Chart`, `ConfigFile`, `CustomResource`, etc., were based on generated code from several years ago, and look a lot different from the what would be generated today. It's a potential source of bugs that would be fixed by recent codegen.
